test/Microsoft.SqlTools.ServiceLayer.IntegrationTests/DisasterRecovery/BackupServiceTests.cs

```diff
@@ -232,6 +232,44 @@ public void ScriptBackupWithAdvancedOptionsTest()
             testDb.Cleanup();
         }
 
+        /// <summary>
+        /// Test the correct script generation for different backup action types
+        /// </summary>
+        [Fact]
+        public void ScriptBackupWithDifferentActionTypesTest()
+        {
+            string databaseName = "testbackup_" + new Random().Next(10000000, 99999999);
+            SqlTestDb testDb = SqlTestDb.CreateNew(TestServerType.OnPrem, false, databaseName);
+            
+            // Create Full backup script
+            string script = GenerateScriptForBackupType(BackupType.Full, databaseName);
+
+            // Validate Full backup script 
+            Assert.Contains("BACKUP DATABASE", script, StringComparison.OrdinalIgnoreCase);
+            Assert.DoesNotContain("BACKUP LOG", script, StringComparison.OrdinalIgnoreCase);
+            Assert.DoesNotContain("DIFFERENTIAL", script, StringComparison.OrdinalIgnoreCase);
+            
+            // Create log backup script
+            script = GenerateScriptForBackupType(BackupType.TransactionLog, databaseName);
+            
+            // Validate Log backup script 
+            Assert.Contains("BACKUP LOG", script, StringComparison.OrdinalIgnoreCase);
+            Assert.DoesNotContain("BACKUP DATABASE", script, StringComparison.OrdinalIgnoreCase);
+            Assert.DoesNotContain("DIFFERENTIAL", script, StringComparison.OrdinalIgnoreCase);
+
+            // Create differential backup script
+            script = GenerateScriptForBackupType(BackupType.Differential, databaseName);
+
+            // Validate differential backup script 
+            Assert.Contains("BACKUP DATABASE", script, StringComparison.OrdinalIgnoreCase);
+            Assert.DoesNotContain("BACKUP LOG", script, StringComparison.OrdinalIgnoreCase);
+            Assert.Contains("WITH  DIFFERENTIAL", script, StringComparison.OrdinalIgnoreCase);
+            
+            // Clean up the database
+            Console.WriteLine("Clean up database..");
+            testDb.Cleanup();
```

Comment:
This isn't going to execute if there's an assertion failure - make sure we always clean up regardless

Reply:
Good point,copied from existing tests and missed this.